pkg/oc/bootstrap/docker/run_self_hosted.go

```diff
@@ -217,61 +240,90 @@ func (c *ClusterUpConfig) BuildConfig() (*configDirs, error) {
 		podManifestDir:               filepath.Join(c.BaseDir, kubelet.PodManifestDirName),
 	}
 
-	if _, err := os.Stat(configLocations.masterConfigDir); os.IsNotExist(err) {
+	originalMasterConfigDir := configs.masterConfigDir
+	originalNodeConfigDir := configs.nodeConfigDir
+
+	if _, err := os.Stat(configs.masterConfigDir); os.IsNotExist(err) {
 		_, err = c.makeMasterConfig()
 		if err != nil {
 			return nil, err
 		}
+		configs.masterConfigDir, err = c.copyToRemote(configs.masterConfigDir, kubeapiserver.KubeAPIServerDirName)
+		if err != nil {
+			return nil, err
+		}
 	}
-	if _, err := os.Stat(configLocations.openshiftAPIServerConfigDir); os.IsNotExist(err) {
-		_, err = c.makeOpenShiftAPIServerConfig(configLocations.masterConfigDir)
+	if _, err := os.Stat(configs.openshiftAPIServerConfigDir); os.IsNotExist(err) {
+		_, err = c.makeOpenShiftAPIServerConfig(originalMasterConfigDir)
+		if err != nil {
+			return nil, err
+		}
+		configs.openshiftAPIServerConfigDir, err = c.copyToRemote(configs.openshiftAPIServerConfigDir, kubeapiserver.OpenShiftAPIServerDirName)
 		if err != nil {
 			return nil, err
 		}
 	}
-	if _, err := os.Stat(configLocations.openshiftControllerConfigDir); os.IsNotExist(err) {
-		_, err = c.makeOpenShiftControllerConfig(configLocations.masterConfigDir)
+	if _, err := os.Stat(configs.openshiftControllerConfigDir); os.IsNotExist(err) {
+		_, err = c.makeOpenShiftControllerConfig(originalMasterConfigDir)
+		if err != nil {
+			return nil, err
+		}
+		configs.openshiftControllerConfigDir, err = c.copyToRemote(configs.openshiftControllerConfigDir, kubeapiserver.OpenShiftControllerManagerDirName)
```

Comment:
@deads2k the reason I have to do this after every step is that some steps depend on files produced by previous step...